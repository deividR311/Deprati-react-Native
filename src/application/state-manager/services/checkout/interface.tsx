import { DeliveryOptionsResponse } from '../../../../infrastucture/apis/delivery-address';
import { Cart } from '../../../../infrastucture/apis/shopping-cart';
import { CheckoutSteps } from '../../../../presentation/navigation/checkout/interfaces';

export interface CheckoutStateHook {
  onContinueButtonTrigger?: CheckoutSteps;
  onPressContinueButton: (screenId: CheckoutSteps) => void;
}

export interface CheckoutState {
  screenId?: CheckoutSteps;
  previousScreen?: CheckoutSteps;
  stackSummary?: any[];
  showLoadingScreen?: boolean;
  onClickedContinueButton: boolean;
  cart: Partial<Cart>;
  totalPriceBeforeSelectDeliveryMode?: string;
  needRefreshCart?: boolean;
  deliveryOptions?: DeliveryOptionsResponse;
  isGiftCardCart: boolean;
  creditInfo?: {
    saveForFuturePay: boolean;
    maskedPhoneNumber: string;
    maskedNumberAccount: string;
    maskedEmail: string;
    directCreditCodeCustomer?: string;
    directCreditCodeAdditional?: string;
  };
  firstTimeSummary?: boolean;
}

export const INITIAL_STATE: CheckoutState = {
  screenId: undefined,
  previousScreen: undefined,
  stackSummary: undefined,
  onClickedContinueButton: false,
  showLoadingScreen: false,
  needRefreshCart: false,
  cart: {},
  deliveryOptions: undefined,
  totalPriceBeforeSelectDeliveryMode: undefined,
  isGiftCardCart: false,
  creditInfo: undefined,
  firstTimeSummary: undefined
};
