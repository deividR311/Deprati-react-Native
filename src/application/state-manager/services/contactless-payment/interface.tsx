import { ContactLessPaymentSteps } from '../../../../presentation/navigation/contactless-payment';

export enum BottomSheetType {
  IDENTITY_VALIDATION = 'IDENTITY_VALIDATION',
  SUCCESS_BUY = 'SUCCESS_BUY'
}

export interface ShowBottomSheetAction {
  bottomSheet: BottomSheetType;
  show: boolean;
  text?: string;
  ticketId?: string;
}

export interface ContactlessPaymentStateHook {
  TicketNumberContactLessPayment?: string;
  showIdentityValidation(show: boolean): void;
  identityValidationIsOpen: boolean;
  showSuccessBuy(show: boolean, text?: string, ticketId?: string): void;
  successBuyIsOpen: boolean;
  successBuyText?: string;
  successBuyTicketId?: string;
  doEnableConfirmPurchaseButton(isEnable: boolean, ticketId?: string): void;
  enableConfirmPurchaseButton: boolean;
}

export interface ContactlessPaymentState {
  tabbar: {
    disableNextButton: boolean;
    disablePreviousButton: boolean;
    currentTab: ContactLessPaymentSteps | '';
  };
  bottomSheetState: {
    identityValidationIsOpen: boolean;
    successBuyIsOpen: boolean;
    successBuyText: string;
    successBuyTicketId: string;
  };
  enableConfirmPurchaseButton: boolean;
}

export const INITIAL_STATE: ContactlessPaymentState = {
  tabbar: {
    currentTab: '',
    disableNextButton: false,
    disablePreviousButton: false
  },
  bottomSheetState: {
    identityValidationIsOpen: false,
    successBuyIsOpen: false,
    successBuyText: '',
    successBuyTicketId: ''
  },
  enableConfirmPurchaseButton: false
};
