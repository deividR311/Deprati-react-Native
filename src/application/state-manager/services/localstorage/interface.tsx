import { NotificationListRes } from '../../../../infrastucture/apis/notifications/notifications.interfaces';
import { MessagesAppInterface } from '../../../../presentation/screens/Splash/hooks/messages.interface';
import { INITIAL_STATE as ContactlessInitialState } from '../contactless-payment';
import {
  UserResponse,
  USER_DEFAULT_DATA
} from '../../../../infrastucture/apis/user/user.interface';

export type LocalStorageParams = Partial<LocalStorageType>;

export interface LocalStorageHook {
  localStorageData: LocalStorageType;
  save(params: LocalStorageParams): void;
  remove(keys: LocalStorageKey[]): void;
  readAndStorageInState(): void;
  fullClean(): void;
}

export enum LocalStorageKey {
  // BEHAVIOR
  WasStoreRead = 'WAS_STORE_READ',
  isAppOpenByFirstTime = 'IS_APP_OPEN_BY_FIRST_TIME',
  IpAddress = 'IP_ADDRESS',
  UID = 'DEVICE_ID',
  IsEnableConfirmPurchaseButtonInQRScreen = 'CONTACTLESS_PAYMENT@IS_ENABLE_CONFIRM_PURCHASE_BUTTON',
  TicketNumberContactLessPayment = 'CONTACTLESS_PAYMENT@TICKET_NUMBER',
  AnonymousCartGuid = 'ANONYMOUS_CART_GUID',

  // USER DATA
  User = 'USER_DATA',
  UserEmail = 'USER_EMAIL',
  UserLastEntry = 'USER_LAST_ENTRY',

  // ACCOUNT DATA
  IsAccountAuthenticated = 'IS_ACCOUNT_AUTHENTICATED',
  AccountNumber = 'ACCOUNT_NUMBER',
  AccountDisplayNumber = 'ACCOUNT_DISPLAY_NUMBER',
  AccountAdditionalNumber = 'ACCOUNT_ADDITIONAL_NUMBER',
  AccountPhoneNumber = 'ACCOUNT_DISPLAY_PHONE_NUMBER',
  LastCustomerId = 'LAST_CUSTOMER_ID',

  //Notifications
  Notifications = 'NOTIFICATIONS',

  //Notifications Push FCM
  TokenFCM = 'TOKENFCM',

  // SECURITY
  Token = 'TOKEN',
  RefreshToken = 'REFRESH_TOKEN',
  ExpiresIn = 'EXPERES_IN',
  ExpiresAt = 'EXPERES_AT',
  GetTokenDate = 'GET_TOKEN_DATE',
  IsLogin = 'IS_LOGIN',
  IsSucessFirstLogin = 'IS_SUCCESS_FIRST_LOGIN',

  BiometricPublicKey = 'publicKey-generate',
  MessagesApp = 'MESSAGES_APP',
  LocalStorageKey = 'LocalStorageKey',
  Whatsapp = 'WHATSAPP'
}

export interface LocalStorageType {
  [LocalStorageKey.WasStoreRead]: boolean;
  [LocalStorageKey.isAppOpenByFirstTime]: boolean;
  [LocalStorageKey.UserLastEntry]: string;
  [LocalStorageKey.UserEmail]: string;
  [LocalStorageKey.AccountNumber]: string;
  [LocalStorageKey.AccountDisplayNumber]: string;
  [LocalStorageKey.AccountAdditionalNumber]: string;
  [LocalStorageKey.AccountPhoneNumber]: string;
  [LocalStorageKey.LastCustomerId]: string;
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]: boolean;
  [LocalStorageKey.TicketNumberContactLessPayment]?: string;
  [LocalStorageKey.AnonymousCartGuid]?: string;
  [LocalStorageKey.User]: UserResponse;
  [LocalStorageKey.Token]: string;
  [LocalStorageKey.RefreshToken]: string;
  [LocalStorageKey.ExpiresIn]: number;
  [LocalStorageKey.ExpiresAt]: string;
  [LocalStorageKey.GetTokenDate]: string;
  [LocalStorageKey.IsLogin]: boolean;
  [LocalStorageKey.IsAccountAuthenticated]: boolean;
  [LocalStorageKey.IpAddress]: string;
  [LocalStorageKey.BiometricPublicKey]: string;
  [LocalStorageKey.UID]: string;
  [LocalStorageKey.IsSucessFirstLogin]: boolean;
  [LocalStorageKey.Notifications]: {
    unread: number;
    notifications: NotificationListRes[];
  };
  [LocalStorageKey.MessagesApp]: MessagesAppInterface;
  [LocalStorageKey.TokenFCM]: string;
  [LocalStorageKey.Whatsapp]: {
    message: string;
    phone: string;
  };
}

export const InitialLocalStorage: LocalStorageType = {
  [LocalStorageKey.WasStoreRead]: false,
  [LocalStorageKey.isAppOpenByFirstTime]: true,
  [LocalStorageKey.AccountNumber]: '',
  [LocalStorageKey.AccountDisplayNumber]: '',
  [LocalStorageKey.AccountAdditionalNumber]: '',
  [LocalStorageKey.AccountPhoneNumber]: '',
  [LocalStorageKey.LastCustomerId]: '',
  [LocalStorageKey.User]: USER_DEFAULT_DATA,
  [LocalStorageKey.UserEmail]: '',
  [LocalStorageKey.UserLastEntry]: '',
  [LocalStorageKey.IsEnableConfirmPurchaseButtonInQRScreen]:
    ContactlessInitialState.enableConfirmPurchaseButton,
  [LocalStorageKey.TicketNumberContactLessPayment]: '',
  [LocalStorageKey.AnonymousCartGuid]: '',
  [LocalStorageKey.Token]: '',
  [LocalStorageKey.RefreshToken]: '',
  [LocalStorageKey.ExpiresIn]: 0,
  [LocalStorageKey.ExpiresAt]: '',
  [LocalStorageKey.GetTokenDate]: '',
  [LocalStorageKey.IsLogin]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IsAccountAuthenticated]: false,
  [LocalStorageKey.IpAddress]: '0.0.0.0',
  [LocalStorageKey.BiometricPublicKey]: '',
  [LocalStorageKey.UID]: '',
  [LocalStorageKey.IsSucessFirstLogin]: false,
  [LocalStorageKey.Notifications]: {
    unread: 0,
    notifications: []
  },
  [LocalStorageKey.MessagesApp]: {},
  [LocalStorageKey.TokenFCM]: '',
  [LocalStorageKey.Whatsapp]: {
    phone: '',
    message: ''
  }
};
