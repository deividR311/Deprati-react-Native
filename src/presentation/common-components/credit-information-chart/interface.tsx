export interface CreditInformationChartProps {
  ownerFullname?: string;
  cardNumber?: string;
  affiliateDate?: string;
  totalAmount?: number;
  amountSpent?: number;
  availableAmount?: number;
  numeroTarjetaAdicionalDisplay?: string;
  showAffiliateDate?: boolean;
  showDisplayCard?: boolean;
  displayCartType?: 'small' | 'large';
  invertValues?: boolean;
  constentChart: any;
  wrapperChartInCard?: boolean;
}

export interface DonutChartProps {
  indicatorValue?: number;
  maxIndicatorValue?: number;
  color?: string;
  dialColor?: string;
  radius?: number;
  strokeWidth?: number;
  duration?: number;
  delay?: number;
  invertValues?: boolean;
}
