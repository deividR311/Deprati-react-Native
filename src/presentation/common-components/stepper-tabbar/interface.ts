import {
  BottomTabBarProps,
  BottomTabNavigationOptions
} from '@react-navigation/bottom-tabs';
import { ParamListBase, TabNavigationState } from '@react-navigation/native';

export interface PropsButtonTab {
  arrowIconDirection: 'left' | 'right';
  options?: BottomTabNavigationOptions;
  isFocused?: boolean;
  onPress?: () => void;
  onLongPress?: () => void;
}

export interface StepperTabBarNavigationProps extends BottomTabBarProps {
  title?: string;
  subtitle?: string;
  showMenu?: () => void;
  onBackFromTabbar?(state: TabNavigationState<ParamListBase>): void;
  onNextFromTabbar?(state: TabNavigationState<ParamListBase>): void;
  showButtomsWhenZero?: boolean;
}
