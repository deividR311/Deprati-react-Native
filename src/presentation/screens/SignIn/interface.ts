export interface CommonDataUserSocialNetwork {
  firsname: string | null;
  lastname: string | null;
  email: string | null;
}
