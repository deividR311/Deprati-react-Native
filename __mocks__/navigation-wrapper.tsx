import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
