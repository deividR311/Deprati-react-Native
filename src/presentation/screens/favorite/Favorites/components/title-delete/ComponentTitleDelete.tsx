import React, { useEffect, useMemo, useState } from 'react';
//Libs
import {
  View,
  Text,
  StyleProp,
  ViewStyle,
  TextStyle,
  TouchableOpacity,
  ActivityIndicator
} from 'react-native';
import PlusSvg from '../../../../../../../assets/icons/Plp/PlusSvg';
import { capitalize } from '../../../../../../application/utils/string-formater';
import { ICON } from '../../../../../../application/utils/enums';
import { COLORS } from '../../../../../../application/common';

interface ComponentTitleDeleteProps {
  title: string;
  style?: StyleProp<ViewStyle>;
  styleText?: StyleProp<TextStyle>;
  styleViewDelete?: StyleProp<ViewStyle>;
  styleViewText?: StyleProp<ViewStyle>;
  sizeDelete?: number;
  onDelete(): void;
  loadingDelete: boolean;
  isItemFavorite?: boolean;
}

const OPACITY = 0.5;

export default function ComponentTitleDelete({
  title,
  onDelete,
  style = {},
  styleText = {},
  styleViewDelete = {},
  styleViewText = {},
  sizeDelete = 14,
  loadingDelete,
  isItemFavorite = false
}: ComponentTitleDeleteProps) {
  const [isRemoving, setIsRemoving] = useState<boolean>(false);

  useEffect(() => {
    if (loadingDelete) return;
    setIsRemoving(false);
  }, [loadingDelete]);

  const handleDelete = () => {
    setIsRemoving(true);
    onDelete();
  };

  const TOUCHABLE_CONFIG = useMemo(() => {
    if (isItemFavorite)
      return {
        activeOpacity: OPACITY,
        hitSlop: {
          top: 23,
          left: 23,
          bottom: 23,
          right: 23
        }
      };
    return { activeOpacity: OPACITY };
  }, [isItemFavorite]);

  return (
    <View style={style}>
      <View style={styleViewText}>
        <Text style={styleText} numberOfLines={2}>
          {capitalize(title)}
        </Text>
      </View>
      <View style={styleViewDelete}>
        <TouchableOpacity
          testID="delete-favorite"
          onPress={handleDelete}
          {...TOUCHABLE_CONFIG}>
          {isRemoving ? (
            <ActivityIndicator
              testID="loading-removing"
              size="small"
              color={COLORS.DARKBRAND}
              style={{ display: isRemoving ? 'flex' : 'none' }}
            />
          ) : (
            <PlusSvg
              testID="icon-delete"
              width={sizeDelete}
              height={sizeDelete}
              rotation={ICON.OPENED}
            />
          )}
        </TouchableOpacity>
      </View>
    </View>
  );
}
