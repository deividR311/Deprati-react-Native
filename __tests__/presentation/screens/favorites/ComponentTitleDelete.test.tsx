import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import { IRender } from '../../../../__mocks__/IRender.interface';
import ComponentTitleDelete from '../../../../src/presentation/screens/favorite/Favorites/components/title-delete/ComponentTitleDelete';

let screenTest: IRender;
const titleProduct: string = 'Camisero h&o';

describe('ComponentTitleDelete Favorites', () => {
  beforeEach(() => {
    screenTest = render(
      <NavigationWrapper
        screen={() => (
          <ComponentTitleDelete
            title={titleProduct}
            onDelete={jest.fn()}
            loadingDelete={false}
          />
        )}
      />
    );
  });

  it('renders correctly ComponentTitleDelete in Snapshot', () => {
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });

  it('Defined title product favorites', () => {
    const { getByText } = screenTest;
    expect(getByText(titleProduct)).toBeDefined();
  });

  it('Defined delete product favorites', () => {
    const { getByTestId } = screenTest;
    const deleteButton = getByTestId('delete-favorite');
    expect(deleteButton).toBeDefined();
  });

  it('action delete product favorites', () => {
    const { getByTestId } = screenTest;
    const deleteButton = getByTestId('delete-favorite');
    expect(deleteButton).toBeDefined();

    fireEvent.press(deleteButton);

    const iconLoading = getByTestId('loading-removing');
    expect(iconLoading).toBeDefined();
  });
});
