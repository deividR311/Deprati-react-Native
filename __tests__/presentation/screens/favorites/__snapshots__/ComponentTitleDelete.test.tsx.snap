// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTitleDelete Favorites renders correctly ComponentTitleDelete in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <View
          style={Object {}}
        >
          <View
            style={Object {}}
          >
            <Text
              numberOfLines={2}
              style={Object {}}
            >
              Camisero h&o
            </Text>
          </View>
          <View
            style={Object {}}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="delete-favorite"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={14}
                bbWidth={14}
                fill="none"
                focusable={false}
                height={14}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 14,
                      "width": 14,
                    },
                  ]
                }
                vbHeight={12}
                vbWidth={12}
                width={14}
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M11.3936 1.81929C11.7285 1.48437 11.7285 0.94135 11.3936 0.606428C11.0586 0.271507 10.5156 0.271507 10.1807 0.606429L6 4.78714L1.81929 0.606429C1.48437 0.271508 0.94135 0.271507 0.606429 0.606429C0.271507 0.94135 0.271507 1.48436 0.606429 1.81929L4.78714 6L0.606429 10.1807C0.271508 10.5156 0.271507 11.0586 0.606428 11.3936C0.94135 11.7285 1.48436 11.7285 1.81929 11.3936L6 7.21286L10.1807 11.3936C10.5156 11.7285 11.0586 11.7285 11.3936 11.3936C11.7285 11.0586 11.7285 10.5156 11.3936 10.1807L7.21286 6L11.3936 1.81929Z"
                    fill={4280693304}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
